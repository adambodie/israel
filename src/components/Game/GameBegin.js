import React from 'react'

const GameBegin = () => (
	<div className='begin'>
		<h2>Press any button to begin</h2>
	</div>
)

export default GameBegin
