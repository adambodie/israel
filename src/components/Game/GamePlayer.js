import React from 'react'

const GamePlayer = ({ index, name, className, buttons, players }) => (
	<div key={index} className='playerResult'>
		<h3>{name}</h3>
		<img className={className} src={`/assets/main/${buttons[players.score].name}.png`} alt={buttons[players.score].name} />
	</div>
)

export default GamePlayer
