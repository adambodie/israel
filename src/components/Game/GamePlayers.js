import React, { Component} from 'react';

export default class GamePlayers extends Component {

	handleChange() {
		this.props.handleClick();
	}
	render() {
		const { result, buttons, playerScore, opponentScore, playerWin, opponentWin, draw } = this.props;
		return (
		<div>
			{this.props.start ? (
				<div className='result'>
					<div className="flex">
						<div>
							<h3>Player</h3>
							<img src={`https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/${buttons[playerScore].name}.png`} alt={buttons[playerScore].name} />
						</div>
						<div>
							<h3>Computer</h3>
							<img src={`https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/${buttons[opponentScore].name}.png`} alt={buttons[opponentScore].name} />
						</div>
					</div>
					<div className="resultText">
						<h4>{result}</h4>
					</div>
					<div className="score">
						<div>
							<h4>Player</h4>
							<h4>{playerWin}</h4>
						</div>
						<div>
							<h4>Draw</h4>
							<h4>{draw}</h4>
						</div>
						<div>
							<h4>Computer</h4>
							<h4>{opponentWin}</h4>
						</div>
					</div>
				</div>
				) : (
				<div className='begin'>
					<h2>Press any button to begin</h2>
				</div>
				)
			}
			</div>
		)
	}
}

