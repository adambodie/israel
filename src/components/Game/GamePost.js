import React from 'react';
import { Row, Column } from 'react-foundation';
import { Link } from 'react-router-dom';

const GamePost = () => {
	return (
		<Row className='display grid-x game-text post'>
			<Column large={6} offsetOnLarge={3}>
				<h1>The Rock Paper Scissors Episode</h1>
				<p>Rock, Paper, Scissors.  A classic game of chance where two people face off and select either Rock, Paper or Scissors.  You all know the rules:  Rock smashes Scissors.  Scissors cuts Paper.  Paper covers Rock.  You either win, you lose, or you tie.  So what does this have to do with the trip?</p>
				<p>Our first activity for the evening, for some reason, was to challenge someone to Rock, Paper, Scissors, with the loser cheering the winner's name.  The winner then challenged another winner all the while your previous opponent chanting your name until there were just two people left to challenge for supremacy.</p>
				<p>I found myself completely going through the motions, not really knowing what the heck was going on, not really paying any attention to the objective of this game or where it would end.  But before I knew it, I somehow found myself in the final match, against Boris, the security guard.  I certainly didn't think much of it, but all I could hear was my name being chanted: 'Adam, Adam, Adam!'  Suddenly, I realized what was going on.</p>
				<p>It went to a Round of Three.  Boris won the first round, but I won the second round.  Could I pull off the Rocky over Drago victory in the final round?  Sadly, no... Boris won, perhaps due to luck, perhaps due to Russian Interference.  But there was no prize money, no medals, no rugelach (I would have gladly taken that!), but I was the People's Champ!  And that, my friend, is why we play Rock, Paper, Scissors here.  Enjoy!</p>
				<div className='prev-next'>
					<Link to={ '/kibbutz' } className='custom-button'>Prev</Link>
					<Link to={ '/cat' } className='custom-button'>Next</Link>
				</div>
			</Column>
		</Row>
	);
};

export default GamePost;
