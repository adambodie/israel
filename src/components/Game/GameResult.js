import React from 'react'

const GameBegin = ({result}) => {
	return (
		<div className='resultText'>
			<h4>{result}</h4>
		</div>
	)
}

export default GameBegin
