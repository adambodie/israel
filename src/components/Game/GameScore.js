import React from 'react'

const GameScore = ({ players, draw }) => {
	return (
		<div className='score'>
			<div>
				<h4>Player</h4>
				<h4>{players[0].win}</h4>
			</div>
			<div>
				<h4>Draw</h4>
				<h4>{draw}</h4>
			</div>
			<div>
				<h4>Computer</h4>
				<h4>{players[1].win}</h4>
			</div>
		</div>
	)
}

export default GameScore
