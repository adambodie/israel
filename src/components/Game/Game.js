import React, { Component} from 'react'
import { Row, Column } from 'react-foundation'
import GamePlayers from './GamePlayers'
import GamePost from './GamePost'

export default class Game extends Component {
	constructor() {
		super()
		this.state = {
			result: '',
			draw: 0,
			start: false,
			buttons: [
				{name: 'rock', id: 0, disabled: false},
				{name: 'paper', id: 1, disabled: false},
				{name: 'scissors', id: 2, disabled: false}
			],
			players: [
				{name: 'Player', score: 0, win: 0, className: ''},
				{name: 'Computer', score: 0, win: 0, className: ''}
			]
		}
		this.handleClick = this.handleClick.bind(this)
		this.handleReset = this.handleReset.bind(this)
	}
	

	handleReset() {
		const newPlayers = [
				{name: 'Player', score: 0, win: 0, className: ''},
				{name: 'Computer', score: 0, win: 0, className: ''}
		]
		this.setState({
			players: newPlayers,
			draw: 0,
			result: '',
			start: false
		})
	}

	handleClick(e, index) {
		setTimeout(() => this.resetClass(), 1500)
		const { buttons, draw, players } = this.state
		const newPlayers = [...players]
		const newButtons = [...buttons]
		newButtons[0].disabled = true
		newButtons[1].disabled = true
		newButtons[2].disabled = true
		newPlayers[1].score = Math.floor(Math.random() * 3)
		newPlayers[0].score = buttons[index].id
		let addDraw = draw
		let newResult = ''
		if (newPlayers[0].score === newPlayers[1].score) {
			newResult = 'It\'s a Draw'
			newPlayers[0].className = 'player-draw'
			newPlayers[1].className = 'computer-draw'
			addDraw++
		}  else if  ((newPlayers[1].score === 2 && newPlayers[0].score === 0) 
					|| (newPlayers[1].score < newPlayers[0].score) ) {
			newResult = `Player wins`
			newPlayers[0].className = 'player-pick'
			newPlayers[1].className = 'loss'
			newPlayers[0].win++
		} else {
			newResult = `Computer wins`
			newPlayers[1].win++
			newPlayers[1].className = 'computer-pick'
			newPlayers[0].className = 'loss'
		}
		this.setState({
			players: newPlayers,
			buttons: newButtons,
			draw: addDraw,
			result: newResult,
			start: true
		})
	}
	
	resetClass() {
		const { players, buttons } = this.state
		const newPlayers = [...players]
		const newButtons = [...buttons]
		newButtons[0].disabled = false
		newButtons[1].disabled = false
		newButtons[2].disabled = false
		newPlayers[1].className = ''
		newPlayers[0].className = ''
		this.setState({
			players: newPlayers,
			buttons: newButtons
		})
	}
	render() {
		const { buttons, players, draw, start, result } = this.state
		return (
			<React.Fragment>
				<Row className='display grid-x game'>
					<Column large={6} offsetOnLarge={3}>
						<h1>Rock, Paper, Scissors!!!</h1>
						<GamePlayers 
							players={players} 
							draw={draw} 
							start={start} 
							handleChange={this.handleClick}
							buttons={buttons}
							result={result} 
							/>
						<div className='buttons'>
							{buttons.map((x, index)=> <button data-testid={`button-${x.name}`} disabled={x.disabled} className={`button-${x.name}`} key={index} onClick={()=> this.handleClick(this, index)}><img src={`/assets/main/${x.name}.png`} alt={x.name}  /></button>)}
							<button className='reset' onClick={()=>this.handleReset()}>Reset</button>
						</div>
					</Column>
				</Row>
				<GamePost />
			</React.Fragment>
		)
	}
}

