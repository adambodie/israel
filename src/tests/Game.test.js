import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Game from '../components/Game/Game';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Game />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const tree = renderer.create(<Game />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders a `h1`', () => {
	const wrapper = shallow(<Game />);
	expect(wrapper.find('h1').length).toEqual(1);
 });

