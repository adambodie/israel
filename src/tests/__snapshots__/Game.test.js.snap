// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row display grid-x game"
>
  <div
    className="columns large-6 large-offset-3"
  >
    <h1>
      Rock, Paper, Scissors!!!
    </h1>
    <div>
      <div
        className="begin"
      >
        <h2>
          Press any button to begin
        </h2>
      </div>
    </div>
    <div
      className="buttons"
    >
      <button
        onClick={[Function]}
      >
        <img
          alt="rock"
          src="https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/rock.png"
        />
      </button>
      <button
        onClick={[Function]}
      >
        <img
          alt="paper"
          src="https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/paper.png"
        />
      </button>
      <button
        onClick={[Function]}
      >
        <img
          alt="scissors"
          src="https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/scissors.png"
        />
      </button>
      <button
        onClick={[Function]}
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
